text/0000-light-dom.md

```diff
@@ -0,0 +1,315 @@
+```
+
+As a result, when the content of a component resides in the Light DOM, it can be accessed like any other content in the Document host, and thus behave like any other content (styling, APIs, accessibility, third party tooling...).
+
+## Motivation
+
+Consumer applications require DOM traversal and observability of an application’s anatomy from the document root. Without this, theming becomes hard and 3rd party applications do not run properly:
+
+- **Theming and branding**
+
+  Because of the CSS isolation, theming is made harder. Typically, theming is done via APIs (CSS properties) and/or CSS theming parts (`::part`). Both come with caveats and issues. Theming is critical for consumer apps, where custom branding is a must have. Some new APIs, like `::theme`, are investigated but they won’t be pervasively available before years.
+  [Styling is critical to web component reuse, but may prove difficult in practice](https://component.kitchen/blog/posts/styling-is-critical-to-web-component-reuse-but-may-prove-difficult-in-practice)
+
+- **Third party integrations**
+
+  Third party tools need to traverse the DOM, which breaks with Shadow DOM and the existing browser APIs (querySelector, ...). Note that the use of Light DOM fixes for Light DOM components, but not for native Shadow DOM ones if the page contains any: Analytics tools, Personalization platforms, Commerce tools like PriceSpider or Honey...
+
+- **Testing software**
+
+  Tools like Selenium, Cypress etc. face the same issues as third party tools when it comes to traversing the DOM.
+
+Introducing components rendering to the Light DOM opens new possibilities where applications can render most of their content in the Light DOM, but also keep individual widgets (e.g. `select` tag) and leaf components that render in the Shadow DOM. This approach combines the best of both worlds by solving the issues presented above and offering strong encapsulation when needed.
+![shadow spectrum](./shadow-spectrum.png?raw=true "Shadow Spectrum")
+
+### Prior Art
+
+Most of the libraries designed to support Shadow DOM also propose a Light DOM option, with a variety of Shadow DOM features (slots, scoped styles, etc.). It includes:
+
+- [StencilJS](https://stenciljs.com/docs/styling#shadow-dom-in-stencil)
+- [LitElement](https://lit-element.polymer-project.org/api/classes/_lit_element_.litelement.html#createrenderroot)
+- [MS Fast Element](https://fast.design/docs/fast-element/working-with-shadow-dom#shadow-dom-configuration)
+
+## Detailed design
+
+### `LightningElement.shadow`
+
+Toggle between light DOM and shadow DOM is done via a new `shadow` static property on the `LightningElement` constructor. This accepts a `boolean` value and is `true` by default. When set to `true`, the LightningElement creates a shadow root on the host element and renders the component content in the shadow root. When set to `false` the component renders its content directly in the host element light DOM.
```

Comment:
When dealing with boolean values, as a best practice, the default should be `false`. It looks like this was maybe modeled after Stencil's API which disables shadow DOM by default, whereas we enable it by default. I would prefer to go with something like `static shadowDisabled = true` or `static disableShadow = true`.

Reply:
Interesting. The idea was to make the `shadow` field accept more complex objects in the future. For example, we would also use the `shadow` field to override the shadow DOM mode:

```js
export default class extends LightningElement {
  static shadow = {
    mode: 'closed'
  };
}
```

FYI @caridy as you were working on something similar to configure LWC components.

Reply:
Agree that defaulting to `true` is weird, but to me the `false` versions feel weirder:

```js
static light = false // "light" what? "light" is an overloaded word
static lightDom = false // verbose, also we now need to choose between lightDom and lightDOM
static noShadow = false // double negative
```

Reply:
Agreed with @nolanlawson on the point above. If we were to change the flag name and use a different mechanism to configure the shadow root, I would vote for @caridy's `static renderToLightDOM: true`.

Reply:
I really only care that the default is `false`, but I'll just point out that the double-negative usually isn't a concern because you don't need to set the property to get the default behavior.

Reply:
After thinking about this some more, I'm ambivalent about `renderToLightDOM`. Although the light DOM doesn't exist in synthetic shadow, both light DOM and shadow DOM do exist in native shadow. We want to indicate that we're going to take what would normally have been rendered in the light DOM and slotted into the shadow DOM, and do some magic without attaching a shadow root. `renderToLightDOM` doesn't seem to capture that, but I don't have any other suggestions at the moment.

Reply:
This is a good point. `shadow = false` implies "no shadow at all", which is true. Whereas `light = false` implies "no light at all," which is not true because it's actually a mix of light and shadow (due to slots).